src/main/java/com/google/ehub/data/ProfileDatastore.java

```diff
@@ -33,22 +38,38 @@ public void addUserProfileToDatastore(String name, String email, String username
     datastore.put(userEntity);
   }
 
+  /**
+   * Create a UserProfile object from a given entity.
+   *
+   * @param userEntity the User Entity in the Datastore
+   * @return a UserProfile object with correct credentials 
+   */
   public UserProfile createUserProfileFromEntity(Entity userEntity) {
-    long id = userEntity.getKey().getId();
     String name = (String) userEntity.getProperty(NAME_PROPERTY_KEY);
     String username = (String) userEntity.getProperty(USERNAME_PROPERTY_KEY);
     String bio = (String) userEntity.getProperty(BIO_PROPERTY_KEY);
     String email = (String) userEntity.getProperty(EMAIL_PROPERTY_KEY);
 
     return new UserProfile(name, username, bio, email);
   }
-  public String getEmailProperty() {
-    return EMAIL_PROPERTY_KEY;
-  }
-  public String getItemKind() {
-    return PROFILE_ITEM_KIND;
-  }
-  public DatastoreService getDatastore() {
-    return datastore;
+
+  /**
+   * Finds a single entity using the email passed in and returns a UserProfile object from the found entity.  
+   * 
+   * @param email the email of the User being added to Datastore
+   * @return a UserProfile object using the entity found
+   */
+  public UserProfile getUserProfile(String email) {
+    if(email == null){
+      return null;
+    }
+    Filter propertyFilter =
+        new FilterPredicate(EMAIL_PROPERTY_KEY, FilterOperator.EQUAL, email);
+    Query query = new Query(PROFILE_ITEM_KIND).setFilter(propertyFilter);
+
+    PreparedQuery queryResults = datastore.prepare(query);
+    Entity userEntity = queryResults.asSingleEntity();
```

Comment:
You should also check if `userEntity == null` here and return null. Otherwise createUserProfileFromEntity could throw a NullPointerException when trying to call `.getProperty(...)` on a null object.